drivers/spi/spi_esp32_spim.c

```diff
@@ -419,7 +332,12 @@ static const struct spi_driver_api spi_api = {
 #define GET_AS_CS(idx)
 #endif
 
+#define SPI2_PINSTATE_IDX	0
+#define SPI3_PINSTATE_IDX	1
```

Comment:
these defines doesn't work properly when board has one SPI instance (like odroid_go only has SPI3).

Reply:
Perhaps related to what @ydamigos is saying here, this does appear to be using instance numbers incorrectly.

Please read https://docs.zephyrproject.org/latest/reference/devicetree/api.html#c.DT_INST carefully, especially these parts:

```
Instance numbers have the following properties:

- for each compatible, instance numbers start at 0 and are contiguous
- exactly one instance number is assigned for each node with a compatible,
  **including disabled nodes**
- enabled nodes (status property is "okay" or missing) are assigned the
  instance numbers starting from 0, and disabled nodes have instance
  numbers which are greater than those of any enabled node

No other guarantees are made. In particular:

- instance numbers **in no way reflect** any numbering scheme that
  might exist in SoC documentation, node labels or unit addresses,
  or properties of the /aliases node (use DT_NODELABEL() or DT_ALIAS()
  for those)
- there **is no general guarantee** that the same node will have
  the same instance number between builds, even if you are building
  the same application again in the same build directory
```

You seem to be making some assumptions here contrary to what the API guarantees. Please adjust the PR accordingly.

Reply:
@mbolivar-nordic yes, the issue I am referring to is related with the wrong SPI instance numbers.   
I tried the following changes based on other SPI driver implementations and it works fine:

```
diff --git a/drivers/spi/spi_esp32_spim.c b/drivers/spi/spi_esp32_spim.c
index ae6d249b27..f018c2331f 100644
--- a/drivers/spi/spi_esp32_spim.c
+++ b/drivers/spi/spi_esp32_spim.c
@@ -327,57 +327,48 @@ static const struct spi_driver_api spi_api = {
 };
 
 #ifdef CONFIG_SOC_ESP32
-#define GET_AS_CS(idx) .as_cs = DT_PROP(DT_NODELABEL(spi##idx), clk_as_cs),
+#define GET_AS_CS(idx) .as_cs = DT_INST_PROP(idx, clk_as_cs),
 #else
 #define GET_AS_CS(idx)
 #endif
 
-#define SPI2_PINSTATE_IDX	0
-#define SPI3_PINSTATE_IDX	1
-
 #define ESP32_SPI_INIT(idx)	\
 				\
-PINCTRL_DT_INST_DEFINE(SPI##idx##_PINSTATE_IDX);	\
+PINCTRL_DT_INST_DEFINE(idx);	\
 										\
 	static struct spi_esp32_data spi_data_##idx = {	\
 		SPI_CONTEXT_INIT_LOCK(spi_data_##idx, ctx),	\
 		SPI_CONTEXT_INIT_SYNC(spi_data_##idx, ctx),	\
-		SPI_CONTEXT_CS_GPIOS_INITIALIZE(DT_NODELABEL(spi##idx), ctx)	\
+		SPI_CONTEXT_CS_GPIOS_INITIALIZE(DT_DRV_INST(idx), ctx)	\
 		.hal = {	\
-			.hw = (spi_dev_t *)DT_REG_ADDR(DT_NODELABEL(spi##idx)),	\
+			.hw = (spi_dev_t *)DT_INST_REG_ADDR(idx),	\
 		},	\
 		.dev_config = {	\
-			.half_duplex = DT_PROP(DT_NODELABEL(spi##idx), half_duplex),	\
+			.half_duplex = DT_INST_PROP(idx, half_duplex),	\
 			GET_AS_CS(idx)							\
-			.positive_cs = DT_PROP(DT_NODELABEL(spi##idx), positive_cs),	\
-			.no_compensate = DT_PROP(DT_NODELABEL(spi##idx), dummy_comp),	\
-			.sio = DT_PROP(DT_NODELABEL(spi##idx), sio)	\
+			.positive_cs = DT_INST_PROP(idx, positive_cs),	\
+			.no_compensate = DT_INST_PROP(idx, dummy_comp),	\
+			.sio = DT_INST_PROP(idx, sio)	\
 		}	\
 	};	\
 		\
 	static const struct spi_esp32_config spi_config_##idx = {	\
-		.spi = (spi_dev_t *)DT_REG_ADDR(DT_NODELABEL(spi##idx)),	\
+		.spi = (spi_dev_t *)DT_INST_REG_ADDR(idx),	\
 			\
-		.clock_dev = DEVICE_DT_GET(DT_CLOCKS_CTLR(DT_NODELABEL(spi##idx))),	\
+		.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(idx)),	\
 		.duty_cycle = 0, \
 		.input_delay_ns = 0, \
-		.irq_source = DT_IRQN(DT_NODELABEL(spi##idx)), \
-		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(SPI##idx##_PINSTATE_IDX),	\
+		.irq_source = DT_INST_IRQN(idx), \
+		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(idx),	\
 		.clock_subsys =	\
-			(clock_control_subsys_t)DT_CLOCKS_CELL(	\
-				DT_NODELABEL(spi##idx), offset),	\
+			(clock_control_subsys_t)DT_INST_CLOCKS_CELL(	\
+				idx, offset),	\
 					\
 	};	\
 		\
-	DEVICE_DT_DEFINE(DT_NODELABEL(spi##idx), &spi_esp32_init,	\
+	DEVICE_DT_INST_DEFINE(idx, &spi_esp32_init,	\
 			      NULL, &spi_data_##idx,	\
 			      &spi_config_##idx, POST_KERNEL,	\
 			      CONFIG_SPI_INIT_PRIORITY, &spi_api);
 
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(spi2), okay)
-ESP32_SPI_INIT(2);
-#endif
-
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(spi3), okay)
-ESP32_SPI_INIT(3);
-#endif
+DT_INST_FOREACH_STATUS_OKAY(ESP32_SPI_INIT)
```